File: PixelDungeonPhotoMaker/src/au/net/genesis/mds/particles/Bubble.java
package au.net.genesis.mds.particles;

import au.net.genesis.mds.helpers.MathHelper;

public class Bubble extends ParticleLogic {
	
	private float scale = MathHelper.randomRange(1F,1.6F);
	
	@Override
	public void update() {
		super.update();
		if (age > lifeTime - 10) {
			alpha = Math.max(alpha-.1F, 0F);
		} else {
			alpha = Math.min(alpha+.2F, 1F);
		}
	}
	
	@Override
	public void initialise() {
		super.initialise();
		alpha = 0;
		yspeed = -0.5F;
		xscale = yscale = scale;
	}
}

File: src/main/java/com/mdsimmo/particles/Bubble.kt
package com.mdsimmo.particles

import com.mdsimmo.helpers.MathHelper

class Bubble : ParticleLogic() {

    override fun newParticle(): ParticleState {
        val scale = MathHelper.randomRange(1f, 1.6f)
        return ParticleState(
            alpha = 0f,
            yspeed = -0.5f,
            yscale = scale,
            xscale = scale
        )
    }

    override fun update(particle: ParticleState): ParticleState {
        return super.update(particle).copy(
                alpha = if (particle.age > particle.lifeTime - 10) {
                    (particle.alpha - .1f).coerceAtLeast(0f)
                } else {
                    (particle.alpha + .2f).coerceAtMost(1f)
                }
        )
    }
}